feat(discord): show player name in rich presence

Klaus-Michealson opened the pull request:
Adds the connected player's Steam name to Discord Rich Presence.

Before:
→ Players 42/128

After:
→ HexaN | 42/128

This gives a more personal and modern look that many servers already use. The change is fully optional (works only if Discord Rich Presence is enabled) and does not affect performance.

Tested on latest Qbox version, no errors.

## Description

<!-- What does your pull request change? Why should it be merged? Does it fix an issue? -->

## Checklist

<!-- Put an x inside the [ ] to check an item, like so: [x] -->

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

Klaus-Michealson committed:
feat(discord): show player name in rich presence

Adds the connected player's Steam name to Discord Rich Presence.

Before:
→ Players 42/128

After:
→ HexaN | 42/128

This gives a more personal and modern look that many servers already use. The change is fully optional (works only if Discord Rich Presence is enabled) and does not affect performance.

Tested on latest Qbox version, no errors.

Klaus-Michealson committed:
update

Displays the connected player's name alongside the player count in Discord Rich Presence.

Before → Players 42/128  
After  → HexaN | 42/128

Common feature requested by the community, used by many popular servers.  
Fully compatible with current config structure and has zero performance impact.

Kenshiin13 committed:
Merge branch 'main' into patch-2

Kenshiin13 reviewed (changes requested):
Can you use config-driven placeholders instead and keep the default behaviour? So every server can decide for themselves what they wish to show in Rich Presence.

`config.richPresence = "{steamName} | {currentPlayers}/{maxPlayers}"`

Kenshiin13 committed:
Merge branch 'main' into patch-2

Kenshiin13 committed:
feat(discord): make rich presence config-driven with placeholders

Klaus-Michealson committed:
Enhance Discord presence with new placeholder checks

-- ESKİ DURUM (Hardcoded & Sabit Yapı)
- local presenceText = ('[%s] %s | %s/%s'):format(playerId, playerName, currentCount or 0, maxPlayers)
- SetRichPresence(presenceText)

-- YENİ DURUM (Dinamik & Config-Driven)
+ local template = discord.richPresence
+ local function render()
+     return (template:gsub('{(%w+)}', {
+         id = usesPlayerId and tostring(GetPlayerServerId(PlayerId())) or nil,
+         charName = usesCharName and getCharName() or nil,
+         playerName = usesPlayerName and GetPlayerName(PlayerId()) or nil,
+         currentPlayers = usesPlayers and (GlobalState.PlayerCount or 0) or nil,
+         maxPlayers = maxPlayers,
+         streetName = usesStreet and getStreetName() or nil,
+     }))
+ end

ChatDisabled committed:
fix(discord): remove Turkish comments and use English fallback

ChatDisabled committed:
docs(discord): document all rich presence placeholders
